src/controllers/wishlistController.js

```diff
@@ -0,0 +1,63 @@
+const mongoose = require('mongoose');
+const Wishlist = require('./wishlist');
+
+// Add an album to a wishlist
+async function addAlbumToWishlist(req, res) {
+  try {
+    const { wishlist_id, album_id } = req.params;
+
+    if (
+      !mongoose.Types.ObjectId.isValid(wishlist_id) ||
+      !mongoose.Types.ObjectId.isValid(album_id)
+    ) {
+      return res.status(400).json({ error: 'Invalid ID format' });
+    }
+
+    const wishlist = await Wishlist.findByIdAndUpdate(
```

Comment:
I have updated function findOneAndUpdate with the suggested parameter , commit e87e86cbbbfc98d977c808a5dd9fb6d23ced9dc9
Akos sorry, but I didn't understand the issue regarding function removeAlbumFromWishlist, could you please give more details?

Reply:
For both `addToWislist` and `removeFromWishlist` handlers, we're fetching the wishlist, and doing the change requested. But at no point do we check if the user who is making this request, is the correct user for the wishlist id that they gave. Of course in the UI this should not be made possible but users may use the API or hackers/malvolent actors may try to spoof requests in the browser.

My suggestion was to change both of the controller handlers to use `findOneAndUpdate` _**instead of** _ `findByIdAndUpdate`. The reason is that `findOneAndUpdate` allows us to give parameters to search by apart from just `id`.

So with `await Wishlist.findOneAndUpdate({ _id: wishlist_id, user: req.user.userId }, ...)`, the result is that the wish list will only be fetched and updated if there is a wishlist with the matching id _that belongs to the requesting user_. Otherwise, it will not find anything to update.

Reply:
Akos , could you check please? commit fdbd9be819f5e0d78c1a3351855b7aca6e40e9c5